GDNative Callable and Signal size mismatch on Android x86

**Godot version:**
`master` (048abb50)


**OS/device including version:**
Fedora 34 compiling for Android x86


**Issue description:**
GDNative build fails for Android x86 in the `master` branch with this error:
```
/root/sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ -o modules/gdnative/gdnative/callable.android.opt.debug.x86.o -c -std=gnu++17 -frtti -fno-exceptions -O2 -fomit-frame-pointer -ftree-vectorize -fpic -ffunction-sections -funwind-tables -fstack-protector-strong -fvisibility=hidden -fno-strict-aliasing -mstackrealign -target i686-none-linux-android -fno-integrated-as -gcc-toolchain /root/sdk/ndk/21.4.7075529/toolchains/x86-4.9/prebuilt/linux-x86_64 -w -Werror=return-type -isystem /root/sdk/ndk/21.4.7075529/sources/cxx-stl/llvm-libc++/include -isystem /root/sdk/ndk/21.4.7075529/sources/cxx-stl/llvm-libc++abi/include --sysroot=/root/sdk/ndk/21.4.7075529/sysroot -isystem /root/sdk/ndk/21.4.7075529/sysroot/usr/include/i686-linux-android -isystem /root/sdk/ndk/21.4.7075529/sources/android/support/include -DNDEBUG -DDEBUG_ENABLED -D__ANDROID_API__=24 -DNO_STATVFS -DGLES_ENABLED -DANDROID_ENABLED -DUNIX_ENABLED -DVULKAN_ENABLED -DNO_FCNTL -DMINIZIP_ENABLED -DZSTD_STATIC_LINKING_ONLY -Imodules/gdnative/include -Ithirdparty/libpng -I/root/sdk/ndk/21.4.7075529/sources/third_party/vulkan/src -I/root/sdk/ndk/21.4.7075529/sources/third_party/vulkan/src/include -I/root/sdk/ndk/21.4.7075529/sources/third_party/vulkan/src/layers -I/root/sdk/ndk/21.4.7075529/sources/third_party/vulkan/src/layers/generated -Ithirdparty/zstd -Ithirdparty/zlib -Iplatform/android -I. modules/gdnative/gdnative/callable.cpp
/root/sdk/ndk/21.4.7075529/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++ -o modules/gdnative/gdnative/signal.android.opt.debug.x86.o -c -std=gnu++17 -frtti -fno-exceptions -O2 -fomit-frame-pointer -ftree-vectorize -fpic -ffunction-sections -funwind-tables -fstack-protector-strong -fvisibility=hidden -fno-strict-aliasing -mstackrealign -target i686-none-linux-android -fno-integrated-as -gcc-toolchain /root/sdk/ndk/21.4.7075529/toolchains/x86-4.9/prebuilt/linux-x86_64 -w -Werror=return-type -isystem /root/sdk/ndk/21.4.7075529/sources/cxx-stl/llvm-libc++/include -isystem /root/sdk/ndk/21.4.7075529/sources/cxx-stl/llvm-libc++abi/include --sysroot=/root/sdk/ndk/21.4.7075529/sysroot -isystem /root/sdk/ndk/21.4.7075529/sysroot/usr/include/i686-linux-android -isystem /root/sdk/ndk/21.4.7075529/sources/android/support/include -DNDEBUG -DDEBUG_ENABLED -D__ANDROID_API__=24 -DNO_STATVFS -DGLES_ENABLED -DANDROID_ENABLED -DUNIX_ENABLED -DVULKAN_ENABLED -DNO_FCNTL -DMINIZIP_ENABLED -DZSTD_STATIC_LINKING_ONLY -Imodules/gdnative/include -Ithirdparty/libpng -I/root/sdk/ndk/21.4.7075529/sources/third_party/vulkan/src -I/root/sdk/ndk/21.4.7075529/sources/third_party/vulkan/src/include -I/root/sdk/ndk/21.4.7075529/sources/third_party/vulkan/src/layers -I/root/sdk/ndk/21.4.7075529/sources/third_party/vulkan/src/layers/generated -Ithirdparty/zstd -Ithirdparty/zlib -Iplatform/android -I. modules/gdnative/gdnative/signal.cpp
modules/gdnative/gdnative/callable.cpp:36:1: error: static_assert failed due to requirement 'sizeof(godot_callable) == sizeof(Callable)' "Callable size mismatch"
static_assert(sizeof(godot_callable) == sizeof(Callable), "Callable size mismatch");
^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
scons: *** [modules/gdnative/gdnative/callable.android.opt.debug.x86.o] Error 1
modules/gdnative/gdnative/signal.cpp:36:1: error: static_assert failed due to requirement 'sizeof(godot_signal) == sizeof(Signal)' "Signal size mismatch"
static_assert(sizeof(godot_signal) == sizeof(Signal), "Signal size mismatch");
^             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
scons: *** [modules/gdnative/gdnative/signal.android.opt.debug.x86.o] Error 1
```


**Steps to reproduce:**
- Build for Android x86 with `scons p=android android_arch=x86 tools=no target=release_debug`

--- a/core/variant/callable.h
+++ b/core/variant/callable.h
@@ -44,9 +44,9 @@ class CallableCustom;
 // is required. It is designed for the standard case (object and method)
 // but can be optimized or customized.
 
+// Enforce 16 bytes with `alignas` to avoid arch-specific alignment issues on x86 vs armv7.
 class Callable {
-	//needs to be max 16 bytes in 64 bits
-	StringName method;
+	alignas(8) StringName method;
 	union {
 		uint64_t object = 0;
 		CallableCustom *custom;
@@ -138,8 +138,9 @@ public:
 // be put inside a Variant, but it is not
 // used by the engine itself.
 
+// Enforce 16 bytes with `alignas` to avoid arch-specific alignment issues on x86 vs armv7.
 class Signal {
-	StringName name;
+	alignas(8) StringName name;
 	ObjectID object;
 
 public:

--- a/modules/gdnative/include/gdnative/callable.h
+++ b/modules/gdnative/include/gdnative/callable.h
@@ -37,6 +37,7 @@ extern "C" {
 
 #include <stdint.h>
 
+// Alignment hardcoded in `core/variant/callable.h`.
 #define GODOT_CALLABLE_SIZE (16)
 
 #ifndef GODOT_CORE_API_GODOT_CALLABLE_TYPE_DEFINED

--- a/modules/gdnative/include/gdnative/signal.h
+++ b/modules/gdnative/include/gdnative/signal.h
@@ -37,6 +37,7 @@ extern "C" {
 
 #include <stdint.h>
 
+// Alignment hardcoded in `core/variant/callable.h`.
 #define GODOT_SIGNAL_SIZE (16)
 
 #ifndef GODOT_CORE_API_GODOT_SIGNAL_TYPE_DEFINED

